pkg/controller/operators/catalog/operator.go

```diff
@@ -1179,18 +1183,41 @@ func (o *Operator) unpackBundles(plan *v1alpha1.InstallPlan) (bool, *v1alpha1.In
 	out := plan.DeepCopy()
 	unpacked := true
 
+	// The bundle timeout annotation if specified overrides the --bundle-unpack-timeout flag value
+	// If the timeout cannot be parsed it's set to < 0 and subsequently ignored
+	unpackTimeout := -1 * time.Minute
+	timeoutStr, ok := plan.GetAnnotations()[bundle.BundleUnpackTimeoutAnnotationKey]
+	if ok {
+		d, err := time.ParseDuration(timeoutStr)
+		if err != nil {
+			o.logger.Errorf("failed to parse unpack timeout annotation(%s: %s): %v", bundle.BundleUnpackTimeoutAnnotationKey, timeoutStr, err)
+		} else {
+			unpackTimeout = d
+		}
+	}
+
 	var errs []error
 	for i := 0; i < len(out.Status.BundleLookups); i++ {
 		lookup := out.Status.BundleLookups[i]
-		res, err := o.bundleUnpacker.UnpackBundle(&lookup)
+		res, err := o.bundleUnpacker.UnpackBundle(&lookup, unpackTimeout)
 		if err != nil {
 			errs = append(errs, err)
 			continue
 		}
 		out.Status.BundleLookups[i] = *res.BundleLookup
 
-		// if pending condition is present, still waiting for the job to unpack to configmap
-		if res.GetCondition(v1alpha1.BundleLookupPending).Status == corev1.ConditionTrue {
+		// if the failed condition is present it means the bundle unpacking has failed
+		failedCondition := res.GetCondition(bundle.BundleLookupFailed)
+		if failedCondition.Status == corev1.ConditionTrue {
+			unpacked = false
+			continue
+		}
+
+		// if the bundle lookup pending condition is present it means that the bundle has not been unpacked
+		// status=true means we're still waiting for the job to unpack to configmap
+		// status=false means the bundle unpack job has failed
```

Comment:
Good catch. This is a hold over from a previous change where I didn't have an explicit condition (`BundleLookupFailed`) for marking the bundle as failed so I was using `BundleLookupPending.status=false` as the failure condition.
I'll revert this back.

Although we never set `BundleLookupPending.status=false` anywhere since we only consider the absence of `BundleLookupPending` condition to indicate that the look up has finished.

So effectively `BundleLookupPending.status=false` and `BundleLookupPending` being missing are the same thing i.e lookup has finished.